Added support to add image to feed

CedricVleminckx opened the pull request:
Add support to add an image to the rss feed

Gummibeer commented:
I agree with @ildyria in all points.

freekmurze commented:
@CedricVleminckx Thanks for this PR, could you also add an example to the readme?

valascus commented:
What happened to this PR? The feature was removed?

CedricVleminckx commented:
Sorry for my inactivity, I will look at this in the weekend

Gummibeer commented on resources/views/rss.blade.php:
```diff
@@ -18,8 +18,8 @@
                 <pubDate>{{ $item->updated->toRssString() }}</pubDate>
-                @if($item->__isset('imageUrl') && $item->__isset('imageSize'))
-                    <enclosure url="{{ $item->imageUrl }}"  type="image/*" length="{{ $item->imageSize }}" />
+                @if($item->__isset('attachmentUrl') && $item->__isset('attachmentSize') && $item->__isset('attachmentType'))
```
Shouldn't the `__isset()` method call be replaceable by a regular `isset()` function call? I think that would be easier to read/more common.

CedricVleminckx replied:
I used this because this is a function in the FeedItem class: https://github.com/spatie/laravel-feed/blob/master/src/FeedItem.php#L149

Gummibeer replied:
This is a PHP magic method which is triggered by calling `isset()` on an inaccessible property.
```php
isset($item->imageUrl)
```
https://www.php.net/manual/en/language.oop5.overloading.php#object.isset

CedricVleminckx replied:
I removed the call to the magic method and used isset().
I also indented some lines for readability

Gummibeer replied:
PS: I think that you should configure your local git or GitHub to use the same email(s). 😉
Right now your commits are unassigned. 😞

ildyria commented on README.md:
```diff
@@ -191,6 +191,62 @@ The `items` key must point to a method that returns one of the following:
+            ->updated($this->updated_at)
+            ->link($this->link)
+            ->author($this->author)
+            ->attachmentUrl($this->attachment_url)
```
I don't see this method in the `FeedItem` class. Am I missing something or some PHP magic?

ildyria commented on README.md:
```diff
@@ -191,6 +191,62 @@ The `items` key must point to a method that returns one of the following:
+            ->link($this->link)
+            ->author($this->author)
+            ->attachmentUrl($this->attachment_url)
+            ->attachmentSize($this->attachment_size)
```
I don't see this method in the `FeedItem` class. Am I missing something or some PHP magic?

ildyria commented on README.md:
```diff
@@ -191,6 +191,62 @@ The `items` key must point to a method that returns one of the following:
+            ->author($this->author)
+            ->attachmentUrl($this->attachment_url)
+            ->attachmentSize($this->attachment_size)
+            ->attachmentType($this->attachment_type)
```
I don't see this method in the `FeedItem` class. Am I missing something or some PHP magic?

CedricVleminckx commented:
Hmm indeed, I didn't notice that. But I can see that the enclosure is only used in the "atom.blade.php" view.
So maybe I should remove the attachment stuff and also render the enclosure in "rss.blade.php" and add validation for the enclosure?

Gummibeer commented:
Am I wrong or does **Atom** does not have an `<enclosure>` or similar?
https://validator.w3.org/feed/docs/atom.html
In opposite to **RSS** which defines the `<enclosure>`.
https://validator.w3.org/feed/docs/rss2.html

CedricVleminckx commented:
You are right. Atom does not have an `<enclosure>` but you can insert a `<link>` with an enclosure: `<link rel="enclosure" href="..." length="..." type="...">`

CedricVleminckx commented:
With the things discussed above.
I removed the attachment stuff and used the already existing enclosure variables and methods.
Then I also changed `rss.blade.php` to include the enclosure and `atom.blade.php` to use the supported link tag with `rel="enclosure"`
Finally I also changed the readme.

freekmurze commented on README.md:
```diff
@@ -191,6 +191,62 @@ The `items` key must point to a method that returns one of the following:
 
+### Enclosures
+
+You can add an enclosure to the feed like so.
```
Could you explain here what an enclosure is?

freekmurze commented on resources/views/atom.blade.php:
```diff
@@ -23,8 +23,12 @@
-            @if($item->__isset('enclosure'))
-              <enclosure url="{{ url($item->enclosure) }}" length="{{ $item->enclosureLength }}" type="{{ $item->enclosureType }}" />
+            @if(
+                isset($item->enclosure)
```
Could we add a method `hasEnclosure` on `$item`, so we don't have to do a compound check here?

Gummibeer commented on src/FeedItem.php:
```diff
@@ -147,6 +147,19 @@ public function validate()
+    public function hasEnclosure()
+    {
+        if (
+            isset($this->enclosure)
```
@freekmurze I would like to know how you would write this `if` regarding your guidelines?
* Keep it as is
* return compound `isset() && ...` directly
* Split into three `if isset()`

In this simple case I think that all options are similar/equal readable. 🤔
But would like to hear your opinion.

valascus commented:
Any progress about this?

bernhardh commented:
I would also love to have this feature

freekmurze commented:
Closing due to inactivity
